texar/modules/decoders/rnn_decoders.py

```diff
@@ -628,9 +626,15 @@ def forward(self,  # type: ignore
             helper, inputs, sequence_length, initial_state,
             max_decoding_length, impute_finished)
 
+        # Release memory and memory_sequence_length in AttentionRNNDecoder
         self.memory = None
         self.memory_sequence_length = None
 
+        # Release memory and memory_sequence_length in AttentionMechanism
+        for attention_mechanism in self._cell._attention_mechanisms:
+            attention_mechanism._values = None
+            attention_mechanism._keys = None
```

Comment:
Nice catch here! For design purposed, I would suggest making these two lines a method of `AttentionMechanism` (called `_clear_cache` or something). This way we'll remember to modify the release code if we add or remove new variables in `AttentionMechanism`. Also, we can document the behavior (and the dependency on `RNNDecoder` to clear the cache) in the docstring of that method.